packages/block-editor/src/components/use-block-drop-zone/index.js

```diff
@@ -130,8 +130,16 @@ export default function useBlockDropZone( {
 
 			setTargetBlockIndex( targetIndex === undefined ? 0 : targetIndex );
 
-			if ( targetIndex !== null ) {
-				showInsertionPoint( targetRootClientId, targetIndex );
+			const nearestBlock = blockElements[ targetIndex ];
+			if ( nearestBlock ) {
+				const previousBlock = blockElements[ targetIndex - 1 ];
+
+				if (
+					nearestBlock.dataset.empty !== 'true' &&
+					previousBlock?.dataset.empty !== 'true'
+				) {
+					showInsertionPoint( targetRootClientId, targetIndex );
+				}
```

Comment:
Good point! It's a side effect of reusing the `useOnBlockDrop` hook here, but it might not be a bad one.

My understanding is that the empty paragraph block itself can be viewed as a dropping indicator since it has little semantic value. But it might not be ideal for some other cases.

This doesn't work great now when dropping a block onto a paragraph block though, the block gets inserted rather than replacing the empty paragraph block. I'll try to fix it or disable it altogether.